VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/General/FlagsEnumValuesAreNotPowersOfTwoTests.cs

```diff
@@ -0,0 +1,1016 @@
+
+        [TestMethod]
+        public void FlagsEnumValuesAreNotPowersOfTwo_ValuesAreNotPowersOfTwo_BaseTypeUInt32_InvokesWarning()
+        {
+            var original = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo : UInt32
+    {
+        Bar = 0,
+        Biz = 1,
+        Baz = 2,
+        Buz = 3,
+        Boz = uint.MaxValue
+    }
+}";
+
+            var result = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo : UInt32
+    {
+        Bar = 0U,
+        Biz = 1U,
+        Baz = 2U,
+        Buz = 4U,
+        Boz = 8U
+    }
+}";
+
+            VerifyDiagnostic(original, string.Format(FlagsEnumValuesAreNotPowersOfTwoAnalyzer.Rule.MessageFormat.ToString(), "Foo"));
+            VerifyFix(original, result);
+        }
+
+        [TestMethod]
+        public void FlagsEnumValuesAreNotPowersOfTwo_ValuesAreNotPowersOfTwo_BaseTypeInt64_InvokesWarning()
+        {
+            var original = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo : Int64
+    {
+        Bar = 0,
+        Biz = 1,
+        Baz = 2,
+        Buz = 3,
+        Boz = long.MaxValue
+    }
+}";
+
+            var result = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo : Int64
+    {
+        Bar = 0L,
+        Biz = 1L,
+        Baz = 2L,
+        Buz = 4L,
+        Boz = 8L
+    }
+}";
+
+            VerifyDiagnostic(original, string.Format(FlagsEnumValuesAreNotPowersOfTwoAnalyzer.Rule.MessageFormat.ToString(), "Foo"));
+            VerifyFix(original, result);
+        }
+
+        [TestMethod]
+        public void FlagsEnumValuesAreNotPowersOfTwo_ValuesAreNotPowersOfTwo_BaseTypeUInt64_InvokesWarning()
+        {
+            var original = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo : UInt64
+    {
+        Bar = 0,
+        Biz = 1,
+        Baz = 2,
+        Buz = 3,
+        Boz = ulong.MaxValue
+    }
+}";
+
+            var result = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo : UInt64
+    {
+        Bar = 0UL,
+        Biz = 1UL,
+        Baz = 2UL,
+        Buz = 4UL,
+        Boz = 8UL
+    }
+}";
+
+            VerifyDiagnostic(original, string.Format(FlagsEnumValuesAreNotPowersOfTwoAnalyzer.Rule.MessageFormat.ToString(), "Foo"));
+            VerifyFix(original, result);
+        }
+
+        [TestMethod]
+        public void FlagsEnumValuesAreNotPowersOfTwo_ValuesArePowersOfTwo_BitShifting_DoesNotInvokeWarning()
+        {
+            var original = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo
+    {
+        Bar = 0,
+        Biz = 1 << 0,
+        Baz = 1 << 1,
+        Buz = 1 << 2,
+        Boz = 1 << 3
+    }
+}";
+
+            VerifyDiagnostic(original);
+        }
+
+        [TestMethod]
+        public void FlagsEnumValuesAreNotPowersOfTwo_ValuesAreNotPowersOfTwo_ValuesOfOtherFlags_DoesNotInvokeWarning()
```

Comment:
This test should result in 

	[Flags]
	enum Days
	{
		None = 0,
		Sunday = 1,
		Monday = 2,
		WorkweekStart = Monday,
		Tuesday = 4,
		Wednesday = 8,
		Thursday = 16,
		Friday = 32,
		WorkweekEnd = Friday,
		Saturday = 64,
		Weekend = Saturday | Sunday,
		Weekdays = Monday | Tuesday | Wednesday | Thursday | Friday
	}

In other words: skip the ones that only have references to other enum values (`WorkWeekStart`, `WorkWeekEnd`, `Weekend` and `Weekdays`) and make the remaining ones flags.